Geometry/CSCGeometryBuilder/plugins/CSCGeometryESModule.cc

```diff
@@ -11,10 +21,16 @@
 
 #include <memory>
 
+#include "Geometry/MuonNumbering/interface/DD4hep_MuonNumbering.h"
+#include "Geometry/MuonNumbering/interface/MuonDDDNumbering.h"
+#include "Geometry/MuonNumbering/interface/MuonBaseNumber.h"
+
 using namespace edm;
 
 CSCGeometryESModule::CSCGeometryESModule(const edm::ParameterSet& p)
-    : alignmentsLabel_(p.getParameter<std::string>("alignmentsLabel")),
+    : useDDD_(p.getParameter<bool>("useDDD")),
+      useDD4hep_{p.getUntrackedParameter<bool>("useDD4hep", false)},
```

Comment:
it has to be tracked, I think

Reply:
Hi Yana, I made the same for GEM migration. It is needed to use Untracked Parameter for dd4hep in order to have no crashes during runTheMatrix tests

Reply:
Strictly speaking tracked vs. untracked should not lead to crashes (or, crashes imply that something else is wrong). Where the failures perhaps complaining a missing parameter? In that case the best solution would be to implement `fillDescriptions()` to generated and validate the configuration fragment.

Reply:
It is not a crash, but test failure 
(i.e ----- Begin Fatal Exception 05-Mar-2020 18:40:27 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing ESProducer: class=CSCGeometryESModule label=''
Exception Message:
MissingParameter: Parameter 'useDD4hep' not found.
----- End Fatal Exception -------------------------------------------------)


 > Strictly speaking tracked vs. untracked should not lead to crashes (or, crashes imply >that something else is wrong). Where the failures perhaps complaining a missing >parameter? In that case the best solution would be to implement `fillDescriptions()` to >generated and validate the configuration fragment.

Reply:
Thanks, so the complaint is about missing parameter. The best solutions are to either implement `fillDescriptions()` (preferred), or propagate the added parameter to all definitions.

Reply:
When you write: "...or propagate the added parameter to all definitions." do you mean to set tUntrackedParameter all the parameters? If it so I think this solution is quite good (and quick) 
> Thanks, so the complaint is about missing parameter. The best solutions are to either >implement `fillDescriptions()` (preferred), or propagate the added parameter to all >definitions.

Reply:
> When you write: "...or propagate the added parameter to all definitions." do you mean to set tUntrackedParameter all the parameters? If it so I think this solution is quite good (and quick)

No, all parameters affecting results/physics should be tracked. I meant to add the parameter in all places where `cms.ESProducer("CSCGeometryESModule", ...)` is defined. By quick look there does not seem to be that many, but it is in the HLT menu, which would need a customize function in https://github.com/cms-sw/cmssw/blob/master/HLTrigger/Configuration/python/customizeHLTforCMSSW.py

Reply:
I found 10 files (see attached picture). In all these .py files do I have to set the dd4hep parameter? Do I have to delete useDD4hep_{p.getUntrackedParameter<bool>("useDD4hep", false)} from CSCGeometryESModule.cc? Probably I'll see a compilation error..
![Screenshot 2020-03-06 19 32 01](https://user-images.githubusercontent.com/28751695/76112022-b0fef280-5fe1-11ea-89e3-b7af2d753e00.png)

Reply:
> In all these .py files do I have to set the dd4hep parameter? 

In the first three (under `Geometry/CSCGeometry*`) yes, the `useDD4hep` parameter would need to be added.

For the next 6 (under `HLTrigger/Configuration`) a customize function should be added in
https://github.com/cms-sw/cmssw/blob/master/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
because the HLT menu dumps should not be edited directly.

The last one looks like it can be ignored (it only modifies an existing module, and the file may well be obsolete).

> Do I have to delete useDD4hep_{p.getUntrackedParameter("useDD4hep", false)} from CSCGeometryESModule.cc?

It should be changed to `useDD4hep_{p.getParameter("useDD4hep")}`.

Reply:
please keep in mind that we have a process modifier https://cmssdt.cern.ch/lxr/source/Configuration/ProcessModifiers/python/dd4hep_cff.py that can be used to centralize settings change. This could help in tracking in a transparent way the set of needed modifications

Reply:
@slomeo I think the easiest way is to set the default value by using `fillDescriptions()`, as suggested by @makortel